pkg/api/v1beta2/defaults.go

```diff
@@ -95,7 +97,43 @@ func init() {
 		},
 		func(obj *Endpoints) {
 			if obj.Protocol == "" {
-				obj.Protocol = "TCP"
+				obj.Protocol = ProtocolTCP
```

Comment:
Can any of this code be factored out between v1beta{1,2,3}? There seems to be quite a bit of duplication?